File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int max = 0;
        int n = height.length;
        // for(int i=0;i<n;i++){
        //     for(int j=i+1;j<n;j++){
        //         max = Math.max(max, (j-i)*Math.min(height[i],height[j]));
        //     }
        // }
        int i=0, j = n-1;
        while(i<j){
            max = Math.max(max, (j-i)*Math.min(height[i],height[j]));
            if(height[i]<height[j]) i++;
            else j--;
        }
        return max;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = (0, len(height) - 1)
        max_area = 0
        while left < right:
            height_left, height_right = (height[left], height[right])
            if height_left < height_right:
                area = (right - left) * height_left
                while height[left] <= height_left and left < right:
                    left += 1
            else:
                area = (right - left) * height_right
                while height[right] <= height_right and left < right:
                    right -= 1
            if area > max_area:
                max_area = area
        return max_area
        